src/mpi/info/info_impl.c

```diff
@@ -271,3 +271,87 @@ int MPIR_Info_create_env_impl(int argc, char **argv, MPIR_Info ** new_info_ptr)
   fn_fail:
     goto fn_exit;
 }
+
+static int hex_encode(char *str, const void *value, int len);
+static int hex_decode(const char *str, void *buf, int len);
+
+int MPIR_Info_set_hex_impl(MPIR_Info * info_ptr, const char *key, const void *value, int value_size)
+{
+    int mpi_errno = MPI_SUCCESS;
+
+    char value_buf[1024];
+    MPIR_Assertp(value_size * 2 + 1 < 1024);
+
+    hex_encode(value_buf, value, value_size);
+
+    mpi_errno = MPIR_Info_set_impl(info_ptr, key, value_buf);
+
+    return mpi_errno;
+}
+
+int MPIR_Info_decode_hex(const char *str, void *buf, int len)
+{
+    int mpi_errno = MPI_SUCCESS;
+
+    int rc = hex_decode(str, buf, len);
+    MPIR_ERR_CHKANDJUMP(rc, mpi_errno, MPI_ERR_OTHER, "**infohexinvalid");
+
+  fn_exit:
+    return mpi_errno;
+  fn_fail:
+    goto fn_exit;
+}
+
+/* ---- internal utility ---- */
+
+/* Simple hex encoding binary as hexadecimal string. For example,
+ * a binary with 4 bytes, 0x12, 0x34, 0x56, 0x78, will be encoded
+ * as ascii string "12345678". The encoded string will have string
+ * length of exactly double the binary size plus a terminating "NUL".
+ */
+
+static int hex_val(char c)
+{
+    if (c >= '0' && c <= '9') {
+        return c - '0';
+    } else if (c >= 'a' && c <= 'f') {
+        return c - 'a' + 10;
+    } else if (c >= 'A' && c <= 'F') {
+        return c - 'A' + 10;
+    } else {
+        return -1;
+    }
+}
```

Comment:
Could you also add something for this function? I gather it is converting ASCII character values back to binary, but it took a while for me to see it.

Reply:
Added https://github.com/pmodels/mpich/blob/0dfaa061000e37e2ff04aabd91b1074031929228/src/mpi/info/info_impl.c#L253